waspc/src/Parser/Auth.hs

```diff
@@ -14,24 +15,52 @@ auth = do
     L.reserved L.reservedNameAuth
     authProperties <- P.waspClosure (L.commaSep1 authProperty)
 
+    let userEntityPs = [s | AuthPropertyUserEntity s <- authProperties]
+    failIfPropMissing propUserEntityName userEntityPs
+
+    let methodsPs = [ms | AuthPropertyMethods ms <- authProperties]
+    failIfPropMissing propMethodsName methodsPs
+
+    let redirectPs = [r | AuthPropertyOnAuthFailedRedirectTo r <- authProperties]
+    failIfPropMissing propOnAuthFailedRedirectToName redirectPs
+
     return Wasp.Auth.Auth
-        { Wasp.Auth._userEntity = getAuthPropUserEntity authProperties
-        , Wasp.Auth._methods = getAuthPropMethods authProperties
+        { Wasp.Auth._userEntity = head userEntityPs
+        , Wasp.Auth._methods = head methodsPs
+        , Wasp.Auth._onAuthFailedRedirectTo = head redirectPs
         }
 
+-- TODO(matija): this should be extracted if we want to use in other places too.
+failIfPropMissing :: (Applicative m, MonadFail m) => String -> [p] -> m ()
+failIfPropMissing propName ps = when (null ps) $ fail errorMsg
```

Comment:
Aha fail, nice that works well in context of Parser, returns line?

Reply:
yes! although it is not the exact line, it is the one of the next closure (I guess it is connected to what amount of input has already been consumed by parser), but still better than nothing.